tests/integration/common-operator-integ-suite.sh

```diff
@@ -151,134 +168,6 @@ build_and_push_image() {
   BUILD_WITH_CONTAINER=0 DOCKER_BUILD_FLAGS=${DOCKER_BUILD_FLAGS} IMAGE=${HUB}/${IMAGE_BASE}:${TAG} TARGET_ARCH=${TARGET_ARCH} make docker-build docker-push
 }
 
-deploy_operator() {
-  echo "Deploying Operator"
-  local TARGET="deploy"
-  if [ "${OCP}" == "true" ]; then
-    # This is a workaround
-    # To avoid errors of certificates meanwhile we are pulling the operator image from the internal registry
-    # We need to set image $HUB to a fixed known value
-    # This value always will be equal to the svc url of the internal registry
-    HUB="image-registry.openshift-image-registry.svc:5000/sail-operator"
-
-    TARGET="deploy-openshift"
-  fi
-  # running docker build inside another container layer causes issues with bind mounts
-  BUILD_WITH_CONTAINER=0 IMAGE=${HUB}/${IMAGE_BASE}:${TAG} make -s --no-print-directory ${TARGET}
-}
-
-check_ready() {
-  local NS=$1
-  local POD_NAME=$2
-  local DEPLOYMENT_NAME=$3
-
-  echo "Check POD: NAMESPACE: \"${NS}\"   POD NAME: \"${POD_NAME}\""
-  timeout --foreground -v -s SIGHUP -k ${TIMEOUT} ${TIMEOUT} bash --verbose -c \
-    "until ${COMMAND} get pod --field-selector=status.phase=Running -n ${NS} | grep ${POD_NAME}; do sleep 5; done"
-
-  echo "Check Deployment Available: NAMESPACE: \"${NS}\"   DEPLOYMENT NAME: \"${DEPLOYMENT_NAME}\""
-  ${COMMAND} wait deployment "${DEPLOYMENT_NAME}" -n "${NS}" --for condition=Available=True --timeout=${TIMEOUT}
-}
-
-logFailure() {
-  echo
-  echo "FAIL: $*"
-}
-
-main_test() {
-  # Add here all the validation tests for the operator
-  echo "Check that istio operator is running"
-  check_ready "${NAMESPACE}" "${DEPLOYMENT_NAME}" "${DEPLOYMENT_NAME}"
-  
-  # Deploy and test every istio version inside versions.yaml
-  versions=$(yq eval '.versions[].name' versions.yaml)
-  echo "Versions to test: ${versions//$'\n'/ }"
-  for ver in ${versions}; do
-    echo "--------------------------------------------------------------"
-    echo "Deploy Istio version '${ver}'"
-    ${COMMAND} get ns "${CONTROL_PLANE_NS}" >/dev/null 2>&1 || ${COMMAND} create namespace "${CONTROL_PLANE_NS}"
-    sed -e "s/version:.*/version: ${ver}/g" "${ISTIO_MANIFEST}" | ${COMMAND} apply -f -
-
-    echo "Wait for Istio to be Reconciled"
-    if ! ${COMMAND} wait "istio/${ISTIO_NAME}" --for condition=Reconciled=True --timeout=${TIMEOUT}; then
-      logFailure "Operator failed to reconcile Istio CR"
-      echo
-      echo "Istio resource:"
-      ${COMMAND} get istio "${ISTIO_NAME}" -n "${CONTROL_PLANE_NS}" -oyaml
-      echo
-      echo "Pods:"
-      ${COMMAND} get pods -n "${CONTROL_PLANE_NS}" -o wide
-      echo
-      echo "Operator logs:"
-      ${COMMAND} logs "deploy/${DEPLOYMENT_NAME}" -n "${NAMESPACE}"
-      exit 1
```

Comment:
operator_test.go is still missing the equivalent of this code. If the test doesn't print this information, we have no idea why the test failed (we only know that the Istio CR didn't become ready, but we have no idea what caused this). This forces us to run the test locally to find the cause of the failure. Of course, the same problem might not occur when running locally, so we really need to make sure that the test prints this information.

Reply:
Added